sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDestinationCheckpointData.cs

```diff
@@ -168,7 +158,11 @@ internal static BlobDestinationCheckpointData Deserialize(Stream stream)
 
             // AccessTier
             JobPlanAccessTier jobPlanAccessTier = (JobPlanAccessTier)reader.ReadByte();
-            AccessTier accessTier = new AccessTier(jobPlanAccessTier.ToString());
+            AccessTier? accessTier = default;
+            if (!jobPlanAccessTier.Equals(JobPlanAccessTier.None))
+            {
+                accessTier = new AccessTier(jobPlanAccessTier.ToString());
+            }
```

Comment:
So `JobPlanAccessTier` is an enum version of the existing extensible enum (struct) `AccessTier`. The existing `AccessTier` does not have a `None` value (well its extensible so it can but I don't think it is intended to) and instead is passed around as nullable. `JobPlanAccessTier.None` maps to when the real `AccessTier` is null so when desterilizing we want to convert it back to null.